### ISSUE TITLE
[bug] basicsr whl requires the future whl too

### ISSUE THREAD

**Issue by melMass** [melMass/comfy_mtb#35]
### Describe the bug

Migrated from #1 by @techoutsider

### Platform and versions

```sh
Windows / Comfy-embedded
```

**Comment by techoutsider** [comment]
im using comfui windows portable, read through the previous thread.

did pip install for requirements and requirements wheels

pip install -r requirements-wheels.txt


D:\2023\OneDrive\klingit\stable_diffusion_xl\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfy_mtb>pip install -r requirements-wheels.txt
Collecting insightface==0.7.3 (from -r requirements-wheels.txt (line 1))
  Using cached insightface-0.7.3.tar.gz (439 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Collecting mmcv==2.0.0 (from -r requirements-wheels.txt (line 2))
  Using cached mmcv-2.0.0.tar.gz (473 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [18 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\joe\AppData\Local\Temp\pip-install-pw4jw4ss\mmcv_8e3bb0d0977742ec832dc2cdc48409d3\setup.py", line 343, in <module>
          ext_modules=get_extensions(),
                      ^^^^^^^^^^^^^^^^
        File "C:\Users\joe\AppData\Local\Temp\pip-install-pw4jw4ss\mmcv_8e3bb0d0977742ec832dc2cdc48409d3\setup.py", line 304, in get_extensions
          ext_ops = extension(
                    ^^^^^^^^^^
        File "C:\Users\joe\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\torch\utils\cpp_extension.py", line 1048, in CUDAExtension
          library_dirs += library_paths(cuda=True)
                          ^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\joe\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\torch\utils\cpp_extension.py", line 1186, in library_paths
          paths.append(_join_cuda_home(lib_dir))
                       ^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\joe\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\torch\utils\cpp_extension.py", line 2223, in _join_cuda_home
          raise EnvironmentError('CUDA_HOME environment variable is not set. '
      OSError: CUDA_HOME environment variable is not set. Please set it to your CUDA install root.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

**Comment by melMass** [comment]
Yes thanks, I know the issue now (it's the title of this issue)

**Comment by melMass** [comment]
Ah no... what you just did is wrong. But what you stated in #1 will help to solve the auto install

**Comment by techoutsider** [comment]
What do you mean? Sorry I’m confused now?

**Comment by melMass** [comment]
Ahah sorry! 
So the reason why there is even a need for all that is that comfy portable doesn't ship with what is required to build some pip dependencies from source (that's normal for portable release to not include "dev" extras).

The reason some pip dependencies need to build from source is that the matrix (your OS, hardware, python version etc) of your system doesn't have "published wheels" i.e prebuilt packages built specifically for your matrix hosted on pip servers. In this case all the build instructions run, but fail in the portable case.

## TLDR
I need to include extra whl files in the next release that I missed as they are not direct dependencies but dependencies of dependencies 😅

**Comment by techoutsider** [comment]
Yes thanks, that’s what I thought, but I also thought I had installed the missing .whl’s.

anyway I’ll wait for the next release.

thanks :)

**Comment by melMass** [comment]
@techoutsider Can you try again, it should have been fixed in https://github.com/melMass/comfy_mtb/releases/tag/v0.1.2

![image](https://github.com/melMass/comfy_mtb/assets/7041726/afc0bfb3-93b3-4fd0-9305-db38208fe491)

**Comment by techoutsider** [comment]
it still gives me the error 


**Total VRAM 10239 MB, total RAM 130980 MB
xformers version: 0.0.20
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 3080 : cudaMallocAsync
Using xformers cross attention
### Loading: ComfyUI-Manager (V0.16)
### ComfyUI Revision: 1216 [fbf5c51c]
Log level: 20
[comfy_mtb] | WARNING -> Web extensions folder at D:\2023\OneDrive\klingit\stable_diffusion_xl\ComfyUI_windows_portable\ComfyUI\web\extensions\mtb is not a symlink, if updating please delete it before
D:\2023\OneDrive\klingit\stable_diffusion_xl\ComfyUI_windows_portable\python_embeded\lib\site-packages\torchvision\transforms\functional_tensor.py:5: UserWarning: The torchvision.transforms.functional_tensor module is deprecated in 0.15 and will be **removed in 0.17**. Please don't rely on it. You probably just need to use APIs in torchvision.transforms.functional or in torchvision.transforms.v2.functional.
  warnings.warn(
RestoreFace
LoadFaceEnhanceModel
[comfy_mtb] | INFO -> Some nodes failed to load:
        Failed to import module faceenhance because ModuleNotFoundError: No module named 'facexlib'

Check that you properly installed the dependencies.
If you think this is a bug, please report it on the github page (https://github.com/melMass/comfy_mtb/issues)
[comfy_mtb] | INFO -> Loaded the following nodes:
        Animation Builder (mtb): Convenient way to manage basic animation maths at the core of many of my workflows
        Smart Step (mtb): Utils to control the steps start/stop of the KAdvancedSampler in percentage
        Text To Image (mtb): Utils to convert text to image using a font
        Styles Loader (mtb): Load csv files and populate a dropdown from the rows (à la A111)
        Bbox From Mask (mtb): From a mask extract the bounding box
        Bbox (mtb): The bounding box (BBOX) custom type used by other nodes
        Crop (mtb): Crops an image and an optional mask to a given bounding box
        Uncrop (mtb): Uncrops an image to a given bounding box
        Debug (mtb): Experimental node to debug any Comfy values, support for more types and widgets is planned
        Save Tensors (mtb): Save torch tensors (image, mask or latent) to disk, useful to debug things outside comfy
        Deep Bump (mtb): Normal & height maps generation from single pictures
        Face Swap (mtb): Face swap using deepinsight/insightface models
        Load Face Swap Model (mtb): Loads a faceswap model
        Load Face Analysis Model (mtb): Loads a face analysis model
        Qr Code (mtb): Basic QR Code generator
        Unsplash Image (mtb): Unsplash Image given a keyword and a size
        String Replace (mtb): Basic string replacement
        Fit Number (mtb): Fit the input float using a source and target range
        Load Film Model (mtb): Loads a FILM model
        Film Interpolation (mtb): Google Research FILM frame interpolation for large motion
        Concat Images (mtb): Add images to batch
        Get Batch From History (mtb): Very experimental node to load images from the history of the server.
        Color Correct (mtb): Various color correction methods
        Image Compare (mtb): Compare two images and return a difference image
        Blur (mtb): Blur an image using a Gaussian filter.
        Mask To Image (mtb): Converts a mask (alpha) to an RGB image with a color and background
        Colored Image (mtb): Constant color image of given size
        Image Premultiply (mtb): Premultiply image with mask
        Image Resize Factor (mtb): Extracted mostly from WAS Node Suite, with a few edits (most notably multiple image support) and less features.
        Save Image Grid (mtb): Save all the images in the input batch as a grid of images.
        Load Image From Url (mtb): Load an image from the given URL
        Save Gif (mtb): Save the images from the batch as a GIF
        Export To Prores (mtb): Export to ProRes 4444 (Experimental)
        Latent Lerp (mtb): Linear interpolation (blend) between two latent vectors
        Image Remove Background Rembg (mtb): Removes the background from the input using Rembg.
        Float To Number (mtb): Node addon for the WAS Suite. Converts a "comfy" FLOAT to a NUMBER.
        Int To Bool (mtb): Basic int to bool conversion
        Int To Number (mtb): Node addon for the WAS Suite. Converts a "comfy" INT to a NUMBER.
        Transform Image (mtb): Save torch tensors (image, mask or latent) to disk, useful to debug things outside comfy
        Load Image Sequence (mtb): Load an image sequence from a folder. The current frame is used to determine which image to load.
        Save Image Sequence (mtb): Save an image sequence to a folder. The current frame is used to determine which image to save.
Total VRAM 10239 MB, total RAM 130980 MB
xformers version: 0.0.20
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 3080 : cudaMallocAsync
WAS Node Suite: OpenCV Python FFMPEG support is enabled
WAS Node Suite Warning: `ffmpeg_bin_path` is not set in `D:\2023\OneDrive\klingit\stable_diffusion_xl\ComfyUI_windows_portable\ComfyUI\custom_nodes\was-node-suite-comfyui\was_suite_config.json` config file. Will attempt to use system ffmpeg binaries if available.
WAS Node Suite: Finished. Loaded 187 nodes successfully.

        "Every great dream begins with a dreamer." - Harriet Tubman**

**Comment by melMass** [comment]
Did you update it? That just the output of launching comfy, on update install should be ran again

**Comment by techoutsider** [comment]
Ah sorry posted the wrong copy pasta, yes I updated it, has the error when installing seemed the same as the one before. 

Will paste it when I’m back at pc

**Comment by melMass** [comment]
I think I know what might happen... the whl are installed in order so basicsr is installed first and tries to build future.. I'll push a fix

**Comment by melMass** [comment]
I pushed the fix I mentioned so on update it should pickup the order now

**Comment by techoutsider** [comment]
i used the comfyui manager, got errors,


#######################################################################
[ComfyUI-Manager] Starting dependency installation/(de)activation for the extension


## Execute install/(de)activation script for 'D:\2023\OneDrive\klingit\stable_diffusion_xl\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfy_mtb'
Ignoring tensorflow: markers 'platform_system != "Windows"' don't match your environment
Collecting gdown@ git+https://github.com/melMass/gdown@main (from -r requirements.txt (line 12))
  Cloning https://github.com/melMass/gdown (to revision main) to c:\users\joe\appdata\local\temp\pip-install-t_9veigd\gdown_cae8a53c39204954a4378a7b5175598b
  Running command git clone --filter=blob:none --quiet https://github.com/melMass/gdown 'C:\Users\joe\AppData\Local\Temp\pip-install-t_9veigd\gdown_cae8a53c39204954a4378a7b5175598b'
  Resolved https://github.com/melMass/gdown to commit 57ea784c863290da4b880c035f99edad854234c3
  Running command git submodule update --init --recursive -q
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting onnxruntime-gpu==1.15.1 (from -r requirements.txt (line 1))
  Obtaining dependency information for onnxruntime-gpu==1.15.1 from https://files.pythonhosted.org/packages/67/4c/03519d7dd781537083b8a0badfc15f6e334b3f7ba227a544860ded5c7157/onnxruntime_gpu-1.15.1-cp310-cp310-win_amd64.whl.metadata
  Using cached onnxruntime_gpu-1.15.1-cp310-cp310-win_amd64.whl.metadata (4.1 kB)
Collecting imageio===2.28.1 (from -r requirements.txt (line 2))
  Using cached imageio-2.28.1-py3-none-any.whl (3.4 MB)
Requirement already satisfied: qrcode[pil] in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from -r requirements.txt (line 3)) (7.4.2)
Requirement already satisfied: numpy==1.23.5 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from -r requirements.txt (line 4)) (1.23.5)
Collecting rembg==2.0.37 (from -r requirements.txt (line 5))
  Obtaining dependency information for rembg==2.0.37 from https://files.pythonhosted.org/packages/8b/37/1a396e156c93fb39638867f8190e8aea52927fbe2484a7da876c0e2cbabe/rembg-2.0.37-py3-none-any.whl.metadata
  Using cached rembg-2.0.37-py3-none-any.whl.metadata (15 kB)
Collecting tensorflow<2.11.0 (from -r requirements.txt (line 7))
  Using cached tensorflow-2.10.1-cp310-cp310-win_amd64.whl (455.9 MB)
Collecting tb-nightly==2.12.0a20230126 (from -r requirements.txt (line 8))
  Using cached tb_nightly-2.12.0a20230126-py3-none-any.whl (5.6 MB)
Collecting protobuf==3.20.2 (from -r requirements.txt (line 11))
  Using cached protobuf-3.20.2-cp310-cp310-win_amd64.whl (904 kB)
Collecting mmdet==3.0.0 (from -r requirements.txt (line 13))
  Using cached mmdet-3.0.0-py3-none-any.whl (1.7 MB)
Collecting facexlib==0.3.0 (from -r requirements.txt (line 14))
  Using cached facexlib-0.3.0-py3-none-any.whl (59 kB)
Requirement already satisfied: coloredlogs in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from onnxruntime-gpu==1.15.1->-r requirements.txt (line 1)) (15.0.1)
Requirement already satisfied: flatbuffers in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from onnxruntime-gpu==1.15.1->-r requirements.txt (line 1)) (23.5.26)
Requirement already satisfied: packaging in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from onnxruntime-gpu==1.15.1->-r requirements.txt (line 1)) (23.1)
Requirement already satisfied: sympy in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from onnxruntime-gpu==1.15.1->-r requirements.txt (line 1)) (1.12)
Requirement already satisfied: pillow>=8.3.2 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from imageio===2.28.1->-r requirements.txt (line 2)) (9.5.0)
Requirement already satisfied: aiohttp>=3.8.1 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from rembg==2.0.37->-r requirements.txt (line 5)) (3.8.4)
Collecting asyncer>=0.0.2 (from rembg==2.0.37->-r requirements.txt (line 5))
  Using cached asyncer-0.0.2-py3-none-any.whl (8.3 kB)
Collecting click>=8.1.3 (from rembg==2.0.37->-r requirements.txt (line 5))
  Obtaining dependency information for click>=8.1.3 from https://files.pythonhosted.org/packages/1a/70/e63223f8116931d365993d4a6b7ef653a4d920b41d03de7c59499962821f/click-8.1.6-py3-none-any.whl.metadata
  Using cached click-8.1.6-py3-none-any.whl.metadata (3.0 kB)
Collecting fastapi>=0.92.0 (from rembg==2.0.37->-r requirements.txt (line 5))
  Obtaining dependency information for fastapi>=0.92.0 from https://files.pythonhosted.org/packages/66/23/5db1adf057fc47f6575c4f317b98882a9312f59a7ec555ae38ca854aef42/fastapi-0.100.1-py3-none-any.whl.metadata
  Using cached fastapi-0.100.1-py3-none-any.whl.metadata (23 kB)
Collecting filetype>=1.2.0 (from rembg==2.0.37->-r requirements.txt (line 5))
  Using cached filetype-1.2.0-py2.py3-none-any.whl (19 kB)
Collecting imagehash>=4.3.1 (from rembg==2.0.37->-r requirements.txt (line 5))
  Using cached ImageHash-4.3.1-py2.py3-none-any.whl (296 kB)
Requirement already satisfied: onnxruntime>=1.14.1 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from rembg==2.0.37->-r requirements.txt (line 5)) (1.15.1)
Requirement already satisfied: opencv-python-headless>=4.6.0.66 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from rembg==2.0.37->-r requirements.txt (line 5)) (4.8.0.74)
Requirement already satisfied: pooch>=1.6.0 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from rembg==2.0.37->-r requirements.txt (line 5)) (1.7.0)
Requirement already satisfied: pymatting>=1.1.8 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from rembg==2.0.37->-r requirements.txt (line 5)) (1.1.8)
Collecting python-multipart>=0.0.5 (from rembg==2.0.37->-r requirements.txt (line 5))
  Using cached python_multipart-0.0.6-py3-none-any.whl (45 kB)
Requirement already satisfied: scikit-image>=0.19.3 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from rembg==2.0.37->-r requirements.txt (line 5)) (0.20.0)
Requirement already satisfied: scipy>=1.9.3 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from rembg==2.0.37->-r requirements.txt (line 5)) (1.11.1)
Requirement already satisfied: tqdm>=4.64.1 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from rembg==2.0.37->-r requirements.txt (line 5)) (4.65.0)
Collecting uvicorn>=0.20.0 (from rembg==2.0.37->-r requirements.txt (line 5))
  Obtaining dependency information for uvicorn>=0.20.0 from https://files.pythonhosted.org/packages/5d/07/b9eac057f7efa56900640a233c1ed63db83568322c6bcbabe98f741d5289/uvicorn-0.23.1-py3-none-any.whl.metadata
  Using cached uvicorn-0.23.1-py3-none-any.whl.metadata (6.2 kB)
Collecting watchdog>=2.1.9 (from rembg==2.0.37->-r requirements.txt (line 5))
  Using cached watchdog-3.0.0-py3-none-win_amd64.whl (82 kB)
Requirement already satisfied: absl-py>=0.4 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from tb-nightly==2.12.0a20230126->-r requirements.txt (line 8)) (1.4.0)
Requirement already satisfied: grpcio>=1.48.2 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from tb-nightly==2.12.0a20230126->-r requirements.txt (line 8)) (1.56.2)
Requirement already satisfied: google-auth<3,>=1.6.3 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from tb-nightly==2.12.0a20230126->-r requirements.txt (line 8)) (2.22.0)
Collecting google-auth-oauthlib<0.5,>=0.4.1 (from tb-nightly==2.12.0a20230126->-r requirements.txt (line 8))
  Using cached google_auth_oauthlib-0.4.6-py2.py3-none-any.whl (18 kB)
Requirement already satisfied: markdown>=2.6.8 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from tb-nightly==2.12.0a20230126->-r requirements.txt (line 8)) (3.4.4)
Requirement already satisfied: requests<3,>=2.21.0 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from tb-nightly==2.12.0a20230126->-r requirements.txt (line 8)) (2.31.0)
Requirement already satisfied: setuptools>=41.0.0 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from tb-nightly==2.12.0a20230126->-r requirements.txt (line 8)) (68.0.0)
Collecting tensorboard-data-server<0.7.0,>=0.6.0 (from tb-nightly==2.12.0a20230126->-r requirements.txt (line 8))
  Using cached tensorboard_data_server-0.6.1-py3-none-any.whl (2.4 kB)
Requirement already satisfied: tensorboard-plugin-wit>=1.6.0 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from tb-nightly==2.12.0a20230126->-r requirements.txt (line 8)) (1.8.1)
Requirement already satisfied: werkzeug>=1.0.1 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from tb-nightly==2.12.0a20230126->-r requirements.txt (line 8)) (2.3.6)
Requirement already satisfied: wheel>=0.26 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from tb-nightly==2.12.0a20230126->-r requirements.txt (line 8)) (0.40.0)
Requirement already satisfied: matplotlib in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from mmdet==3.0.0->-r requirements.txt (line 13)) (3.7.2)
Collecting pycocotools (from mmdet==3.0.0->-r requirements.txt (line 13))
  Using cached pycocotools-2.0.6.tar.gz (24 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting shapely (from mmdet==3.0.0->-r requirements.txt (line 13))
  Using cached shapely-2.0.1-cp310-cp310-win_amd64.whl (1.4 MB)
Requirement already satisfied: six in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from mmdet==3.0.0->-r requirements.txt (line 13)) (1.16.0)
Collecting terminaltables (from mmdet==3.0.0->-r requirements.txt (line 13))
  Using cached terminaltables-3.1.10-py2.py3-none-any.whl (15 kB)
Collecting filterpy (from facexlib==0.3.0->-r requirements.txt (line 14))
  Using cached filterpy-1.4.5.zip (177 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\joe\AppData\Local\Temp\pip-install-t_9veigd\filterpy_9f155f7a1d074090bc0fdaf5df6f0135\setup.py", line 4, in <module>
          import filterpy
      ModuleNotFoundError: No module named 'filterpy'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
install/(de)activation script failed: D:\2023\OneDrive\klingit\stable_diffusion_xl\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfy_mtb

## Execute install/(de)activation script for 'D:\2023\OneDrive\klingit\stable_diffusion_xl\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfy_mtb'
[mtb install]  No arguments provided, doing a full install/update...
[mtb install]  Detected environment: embeded
[mtb install]  Installing dependencies from requirements.txt...
[mtb install]  Package onnxruntime-gpu==1.15.1 already installed (import name: 'onnxruntime').
[mtb install]  Package imageio===2.28.1 already installed (import name: 'imageio').
[mtb install]  Package qrcode already installed (import name: 'qrcode').
[mtb install]  Package numpy==1.23.5 already installed (import name: 'numpy').
[mtb install]  Package rembg==2.0.37 already installed (import name: 'rembg').
[mtb install]  Package tensorflow<2.11.0 already installed (import name: 'tensorflow').
[mtb install]  Package tb-nightly==2.12.0a20230126 already installed (import name: 'tensorboard').
[mtb install]  Package tensorflow already installed (import name: 'tensorflow').
[mtb install]  Package protobuf==3.20.2 already installed (import name: 'google.protobuf').
[mtb install]  Package gdown already installed (import name: 'gdown').
[mtb install]  Installing package mmdet==3.0.0...
Collecting mmdet==3.0.0
  Using cached mmdet-3.0.0-py3-none-any.whl (1.7 MB)
Requirement already satisfied: matplotlib in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from mmdet==3.0.0) (3.7.2)
Requirement already satisfied: numpy in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from mmdet==3.0.0) (1.23.5)
Collecting pycocotools (from mmdet==3.0.0)
  Using cached pycocotools-2.0.6.tar.gz (24 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: scipy in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from mmdet==3.0.0) (1.11.1)
Collecting shapely (from mmdet==3.0.0)
  Using cached shapely-2.0.1-cp310-cp310-win_amd64.whl (1.4 MB)
Requirement already satisfied: six in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from mmdet==3.0.0) (1.16.0)
Collecting terminaltables (from mmdet==3.0.0)
  Using cached terminaltables-3.1.10-py2.py3-none-any.whl (15 kB)
Requirement already satisfied: contourpy>=1.0.1 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from matplotlib->mmdet==3.0.0) (1.1.0)
Requirement already satisfied: cycler>=0.10 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from matplotlib->mmdet==3.0.0) (0.11.0)
Requirement already satisfied: fonttools>=4.22.0 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from matplotlib->mmdet==3.0.0) (4.41.1)
Requirement already satisfied: kiwisolver>=1.0.1 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from matplotlib->mmdet==3.0.0) (1.4.4)
Requirement already satisfied: packaging>=20.0 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from matplotlib->mmdet==3.0.0) (23.1)
Requirement already satisfied: pillow>=6.2.0 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from matplotlib->mmdet==3.0.0) (9.5.0)
Requirement already satisfied: pyparsing<3.1,>=2.3.1 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from matplotlib->mmdet==3.0.0) (3.0.9)
Requirement already satisfied: python-dateutil>=2.7 in d:\2023\onedrive\klingit\stable_diffusion_xl\comfyui_windows_portable\python_embeded\lib\site-packages (from matplotlib->mmdet==3.0.0) (2.8.2)
Building wheels for collected packages: pycocotools
  Building wheel for pycocotools (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for pycocotools (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [17 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-310
      creating build\lib.win-amd64-cpython-310\pycocotools
      copying pycocotools\coco.py -> build\lib.win-amd64-cpython-310\pycocotools
      copying pycocotools\cocoeval.py -> build\lib.win-amd64-cpython-310\pycocotools
      copying pycocotools\mask.py -> build\lib.win-amd64-cpython-310\pycocotools
      copying pycocotools\__init__.py -> build\lib.win-amd64-cpython-310\pycocotools
      running build_ext
      D:\2023\OneDrive\klingit\stable_diffusion_xl\ComfyUI_windows_portable\python_embeded\lib\site-packages\Cython\Compiler\Main.py:381: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed from earlier releases! File: C:\Users\joe\AppData\Local\Temp\pip-install-gmw3zj8y\pycocotools_4822544ce6194348bf7e088babf3fb70\pycocotools\_mask.pyx
        tree = Parsing.p_module(s, pxd, full_module_name)
      Compiling pycocotools/_mask.pyx because it changed.
      [1/1] Cythonizing pycocotools/_mask.pyx
      building 'pycocotools._mask' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pycocotools
Failed to build pycocotools
ERROR: Could not build wheels for pycocotools, which is required to install pyproject.toml-based projects
[mtb install]  Failed to install package mmdet==3.0.0 using pip package name  (import name: 'mmdet'). Error: Command '['D:\\2023\\OneDrive\\klingit\\stable_diffusion_xl\\ComfyUI_windows_portable\\python_embeded\\python.exe', '-m', 'pip', 'install', 'mmdet==3.0.0']' returned non-zero exit status 1.
[mtb install]  Installing package facexlib==0.3.0...
Collecting facexlib==0.3.0
  Using cached facexlib-0.3.0-py3-none-any.whl (59 kB)
Collecting filterpy (from facexlib==0.3.0)
  Using cached filterpy-1.4.5.zip (177 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\joe\AppData\Local\Temp\pip-install-qeuk4voq\filterpy_ce7cbf37ae2142c4a8c1caf6e380df37\setup.py", line 4, in <module>
          import filterpy
      ModuleNotFoundError: No module named 'filterpy'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
[mtb install]  Failed to install package facexlib==0.3.0 using pip package name  (import name: 'facexlib'). Error: Command '['D:\\2023\\OneDrive\\klingit\\stable_diffusion_xl\\ComfyUI_windows_portable\\python_embeded\\python.exe', '-m', 'pip', 'install', 'facexlib==0.3.0']' returned non-zero exit status 1.
[mtb install]  Downloading and installing release wheels since we are in a Comfy embeded environment
[mtb install]  Downloading and installing release wheels since no arguments where provided
[mtb install]  Package insightface==0.7.3 already installed (import name: 'insightface').
[mtb install]  Package mmcv==2.0.0 already installed (import name: 'mmcv').
D:\2023\OneDrive\klingit\stable_diffusion_xl\ComfyUI_windows_portable\python_embeded\lib\site-packages\torchvision\transforms\functional_tensor.py:5: UserWarning: The torchvision.transforms.functional_tensor module is deprecated in 0.15 and will be **removed in 0.17**. Please don't rely on it. You probably just need to use APIs in torchvision.transforms.functional or in torchvision.transforms.v2.functional.
  warnings.warn(
[mtb install]  Package basicsr==1.4.2 already installed (import name: 'basicsr').
[mtb install]  All required wheels are already installed.

[ComfyUI-Manager] Startup script completed.
#######################################################################

**Comment by melMass** [comment]
Should be fixed in https://github.com/melMass/comfy_mtb/releases/tag/v0.1.3

**Comment by techoutsider** [comment]
Nice with a quick check,  it seems to be working!

**Comment by techoutsider** [comment]
I did notice in your node list there was colour nodes, hsv - rgb and rgb - hsv, which i dont appear to have, but other than that i dont seem to have any errors :)

**Comment by melMass** [comment]
That's expected! The node list is outdated, I think I only removed the few you mentioned, but added others that are not yet properly documented!

I will keep this issue open until I merge #41

**Comment by techoutsider** [comment]
Shame seemed like a nice node to have, but then everything is working as expected!

/status is nice!!!
